torch/csrc/jit/codegen/cuda/lower_index_compute.cpp

```diff
@@ -799,11 +799,12 @@ IndexFromIdGraph getTensorIndexFromIdGraph(
   indexing.run(loop_indexing);
 
   // Populate indexing through exact map from initial indexing
+  auto consumer_root = index_producer ? consumer_tv->getRootDomain()
+                                      : consumer_tv->getMaybeRFactorDomain();
 
   // First collect all iterdomains in consumer transform history.
   auto all_consumer_vals = DependencyCheck::getAllValsBetween(
-      {consumer_tv->getMaybeRFactorDomain().begin(),
-       consumer_tv->getMaybeRFactorDomain().end()},
+      {consumer_root.begin(), consumer_root.end()},
```

Comment:
Haven't caught up with the new indexing changes, so tagging @shmsong

Reply:
Yeah, I was quite surprised we didn't index consumers all the way back to roots. This was a bug as far as I'm concerned in indexing.